src/Runner.Worker/Handlers/StepHost.cs

```diff
@@ -16,18 +19,20 @@ public interface IStepHost : IRunnerService
         event EventHandler<ProcessDataReceivedEventArgs> OutputDataReceived;
         event EventHandler<ProcessDataReceivedEventArgs> ErrorDataReceived;
 
-        string ResolvePathForStepHost(string path);
+        string ResolvePathForStepHost(IExecutionContext executionContext, string path);
```

Comment:
Should we add `IExecutionContext` prop to `IStepHost` interface or `IDefaultStepHost` because in `ContainerStepHost` class we already have `ContainerInfo` prop which will be redundant when we add `IExecutionContext` prop to it?

Reply:
I think this is a bigger refactoring effort, it might obfuscate the PR a bit too much.

After we sort out redundancies like that ☝️ though I'd be more than happy to take a look

Reply:
I like this idea, but we can take it as a refactor item after this pr merges